src/main/scala/TileResetCtrl.scala

```diff
@@ -61,7 +61,8 @@ class TLTileResetCtrl(w: Int, params: TileResetCtrlParams, tile_prci_domains: Se
             // clocked to the bus this is attached to, not the clock in this
             // clock bundle. We expect a ClockGroupResetSynchronizer downstream
             // to synchronize the resets
-            oD.reset := r.asBool.asAsyncReset
+            // Also, this or enforces that the tiles come out of reset after the reset of the system
+            oD.reset := (r.asBool || asyncResetSinkNode.in.head._1.reset.asBool).asAsyncReset
```

Comment:
How does this enforce your claim? The register is reset by the asyncReset and then can only be changed by something else in the system (which would need to be out of reset).
If you want to enforce a delay you should actually put a delay on the reset IMO.

Reply:
This async reset reg isn't necessarily init=1. In the case where init=0, we still need to have the time receive the async reset.

Reply:
Ok. I see now. In that past this hasn't been an issue since the `init=0` or always on blocks just directly receive the off-chip reset without a register to control them.

Reply:
Since the SCR reg. is by default set to `0`, then the core is out of reset before the rest of the system in the current setup. By adding the `||` you enforce that the "global" async reset is 1st de-asserted before the async SCR register has any effect (at that point you toggle the tile reset). This isn't an issue of delay because the initial state of the SCR register is `0`.